backend/app2/services/user_service.py

```
@@ -140,3 +142,78 @@ async def change_password(
+        storage_path = f"{user_id}/avatar_{uuid.uuid4()}.{file_extension}"
+        bucket_name = "profile-pics"
+
+        try:
+            logger.info(f"Uploading avatar to Supabase Storage. Bucket: {bucket_name}, Path: {storage_path}")
+            
+            # Read file content
+            file_content = await file.read()
+            
+            # Upload to Supabase Storage
+            # The Supabase Python client's `upload` method expects bytes.
+            upload_response = supabase.client.storage.from_(bucket_name).upload(
```

Ruff (0.11.9)

Local variable `upload_response` is assigned to but never used

Remove assignment to unused variable `upload_response`

(F841)